docs/source/markdown/podman-network-create.1.md

```diff
@@ -63,6 +67,19 @@ Create a network named *newnet* that uses *192.5.0.0/16* for its subnet.
 /etc/cni/net.d/newnet.conflist
 ```
 
+Create a dual-stack network named *newDualnet* that uses an IPv6 range *2001:db8::/64* for its subnet.
+The IPv4 range is autogenerated.
```

Comment:
Why use `2001:db8::/64?   Can't the IPv6 range can be autogenerated too?  Preferably from a ULA and not a global range that someday might belong to someone?

Reply:
this is the current docker behaviour, you have to specify the IPv6 range to create the network, the `2001:db8::/64`is just for documentation purposes, the user can specify any range meanwhile there is no conflict

Reply:
For docs it makes sense to use [IPv6 Address Prefix Reserved for Documentation](https://tools.ietf.org/html/rfc3849)

When using create, maybe generate IPv6 prefix (from ULA) when the prefix is not supplied by user?  I know docker doesn't do this but docker also doesn't have rootless either.  We can improve on docker.     Otherwise unless a prefix is provided to the users, they'll just take `2001:db8::/64` because it was in the doc.

Today, for IPv4, every podman system uses [10.88.0.0/16](https://github.com/containers/podman/blob/master/cni/87-podman-bridge.conflist).  I'm suggesting that this conflict be avoided with IPv6.

Reply:
We should probably add a default-prefix field in containers.conf for storing a v6 prefix to use (and, for that matter, an IPv4 prefix - I believe we are presently hardcoding our auto-assigned v4 prefixes to be the first open /24 in 10.89.0.0/16)

Reply:
When `containers.conf` is created for the first time, I suggest to use a [Pseudo-Random Global ID](https://tools.ietf.org/html/rfc4193#section-3.2.2) to pick the prefix.  The Subnet ID for each network can pick the first open value from there.

If user re-install, they get new ULA, which is what happens with CPE gear that follow [RFC7084](https://tools.ietf.org/html/rfc7084) get factory reset.

Reply:
so, how https://podman.io/blogs/2020/07/01/rest-versioning.html applies here?

> Podman’s REST API is split into two halves: one providing a Docker-compatible API, and a Libpod API providing support for Podman’s unique features such as pods.

`podman network create` has to be docker compatible or no? because if it does and we modify the behaviour to assign an IPv6 when docker fails, we are breaking the compatibility (I'm not entering the discussion now on which one is better, just which is correct )

Reply:
The CLI is not strictly required to be Docker compatible, but we try to match them wherever possible for consistency and ease of migration; when we break compatibility, we should do so with good reason.

The REST API's compat endpoints are required to be strictly Docker compatible.

Reply:
I personally think that changing the defaults or the behaviour respect docker should not be on the scope of this PR, since that will require more discussion, i.e, Michael suggestion is really interesting https://github.com/containers/podman/pull/8143#discussion_r512126735 but that differs from IPv4 behaviour, where you always have a well-known-subnet and most CIs and user leverage (is kind of an API right now)

Reply:
yes lets talk about changes of defaults in a different PR.  I am fine with adding something to containers.conf to implement the change.  If @mccv1r0 defaults is a better idea, we could switch to that default and then allow users to switch containers.conf into Docker mode.

Reply:
Wait, do we use the same /64 for *every* new v6 network by default? I think we do need some logic to increment that if we already have a network with that subnet, otherwise we wait until CNI tries to start a container in the network to blow up with an error...

Reply:
Current pr implement the docker behaviour, there is no defaulting for ipv subnets https://github.com/containers/podman/pull/8143#discussion_r512068637

I'm happy to keep contributing and follow up this discussion in a new pr/issue/meeting
https://github.com/containers/podman/pull/8143#discussion_r512142451